File: FinancialGaming/src/game/gameConstants.js
export const TOTAL_ROUNDS = 4;

export const STARTING_CASH = 2_400_000;
export const STARTING_LOAN = 1_500_000;
export const STARTING_CREDIT_SCORE = 500;
export const MIN_CREDIT_SCORE_TO_BORROW = 200;

export const SALARY_PER_ROUND = 2_400_000;
export const SURVIVAL_COST = 700_000;
export const LOAN_INTEREST_RATE = 0.1;

/** Minimum loan payment — increases 10% each round (round 1 base) */
export const BASE_MIN_LOAN_PAYMENT = 150_000;

export const getMinimumLoanPayment = (round) =>
  Math.round(BASE_MIN_LOAN_PAYMENT * Math.pow(1 + LOAN_INTEREST_RATE, round - 1));

export const HOUSING_OPTIONS = {
  PARENTS: {
    id: "PARENTS",
    name: "Stay with Parent/Guardian",
    baseCost: 150_000,
    rule: "Inflation @2% on dice",
    inflationOnDice: 0.02,
  },
  SHARED: {
    id: "SHARED",
    name: "Shared Apartment",
    baseCost: 300_000,
    rule: "20% Fixed",
    fixedRate: 0.2,
  },
  SINGLE: {
    id: "SINGLE",
    name: "Single Apartment",
    baseCost: 900_000,
    rule: "15% Fixed",
    fixedRate: 0.15,
  },
  LUXURY: {
    id: "LUXURY",
    name: "Luxury Apartment",
    baseCost: 1_500_000,
    rule: "5% Fixed",
    fixedRate: 0.05,
  },
};

export const DICE_EVENTS = {
  1: {
    name: "Job Loss",
    description: "Miss next round's salary",
    type: "PENALTY",
  },
  2: {
    name: "Medical Emergency",
    cost: 400_000,
    type: "PENALTY",
  },
  3: {
    name: "Family Emergency",
    cost: 300_000,
    type: "PENALTY",
  },
  4: {
    name: "Family Support",
    gain: 200_000,
    type: "GAIN",
  },
  5: {
    name: "Investment Opportunity",
    minBalance: 500_000,
    cost: 500_000,
    payout: 900_000,
    type: "INVESTMENT",
    description: "Pay ₦500k if balance ≥ ₦500k — receive ₦900k once",
  },
  6: {
    name: "Investment Opportunity",
    minBalance: 600_000,
    cost: 600_000,
    recurringPayout: 340_000,
    type: "INVESTMENT",
    description: "Pay ₦600k if balance ≥ ₦600k — receive ₦340k at start of each remaining round",
  },
};

/** In-round phases only — housing is pre-game setup, not a round phase */
export const PHASES = {
  SURVIVAL: 1,
  LOAN: 2,
  DICE: 3,
  NETWORTH: 4,
};

export const GAME_STAGES = {
  LOBBY: "lobby",
  HOUSING: "housing",
  PLAYING: "playing",
  COMPLETE: "complete",
};

/** Phases with no player input — all players resolve at the same time */
export const SIMULTANEOUS_PHASES = [PHASES.NETWORTH];

/** Pause after an emergency dice roll before advancing to the next player */
export const DICE_RESULT_DELAY_MS = 4000;

export const NET_WORTH_PHASE_DURATION_MS = 5000;

/** How long round-start payouts (salary, etc.) are shown to all players */
export const ROUND_START_DURATION_MS = 60_000;

export const isSimultaneousPhase = (phase) => SIMULTANEOUS_PHASES.includes(phase);

File: FinancialGamingBackend/src/main/java/com/financegaming/engine/GameConstants.java
package com.financegaming.engine;

public final class GameConstants {

    private GameConstants() {
    }

    public static final int TOTAL_ROUNDS = 4;
    public static final long STARTING_CASH = 2_400_000L;
    public static final long STARTING_LOAN = 1_500_000L;
    public static final int STARTING_CREDIT_SCORE = 500;
    public static final int MIN_CREDIT_SCORE_TO_BORROW = 200;
    public static final int MIN_CREDIT_SCORE = 300;
    public static final int MAX_CREDIT_SCORE = 850;
    public static final long SALARY_PER_ROUND = 2_400_000L;
    public static final long SURVIVAL_COST = 700_000L;
    public static final double LOAN_INTEREST_RATE = 0.1;
    public static final long BASE_MIN_LOAN_PAYMENT = 150_000L;
    public static final int PLAYER_COUNT = 3;

    public static long getMinimumLoanPayment(int round) {
        return Math.round(BASE_MIN_LOAN_PAYMENT * Math.pow(1 + LOAN_INTEREST_RATE, round - 1));
    }

    public record HousingOption(long baseCost, double fixedRate, double inflationOnDice) {
    }

    public static HousingOption housingOption(com.financegaming.domain.HousingType type) {
        return switch (type) {
            case PARENTS -> new HousingOption(150_000L, 0, 0.02);
            case SHARED -> new HousingOption(300_000L, 0.2, 0);
            case SINGLE -> new HousingOption(900_000L, 0.15, 0);
            case LUXURY -> new HousingOption(1_500_000L, 0.05, 0);
        };
    }

    public record DiceEventDef(
            String name,
            String description,
            String type,
            Long cost,
            Long gain,
            Long minBalance,
            Long payout,
            Long recurringPayout
    ) {
    }

    public static DiceEventDef diceEvent(int roll) {
        return switch (roll) {
            case 1 -> new DiceEventDef("Job Loss", "Miss next round's salary", "PENALTY", null, null, null, null, null);
            case 2 -> new DiceEventDef("Medical Emergency", null, "PENALTY", 400_000L, null, null, null, null);
            case 3 -> new DiceEventDef("Family Emergency", null, "PENALTY", 300_000L, null, null, null, null);
            case 4 -> new DiceEventDef("Family Support", null, "GAIN", null, 200_000L, null, null, null);
            case 5 -> new DiceEventDef("Investment Opportunity", null, "INVESTMENT", 500_000L, null, 500_000L, 900_000L, null);
            case 6 -> new DiceEventDef("Investment Opportunity", null, "INVESTMENT", 600_000L, null, 600_000L, null, 340_000L);
            default -> null;
        };
    }
}
